### PR TITLE
Bootstrap local dev environment

### PR DESCRIPTION
**Pull request by filewalkwithme**
This PR adds scripts to help a contributor to launch a local `dev` environment.

### :warning: Note: this need to be reviewed and merged after #33 :warning:

There are 3 scripts included:

- `contrib/dev/setup-dependencies.sh`: setup a postgres database and a keycloak server (SAML provider)
- `contrib/dev/run.sh`: build and run the Vulcan API
- `contrib/dev/teardown.sh`: destroy all containers created by `setup-dependencies.sh`

I have added a new section to the `README.md` containing detailed instructions on how to run these scripts.

Sample output:
```
./setup-dependencies.sh 
Setting up Keycloak...

730d7f37cd17982c41fcc37e44175d734858130aacdb6109331cac686fa0eb3a
--> Keycloak ready.

Setting up postgres...

Creating vulcanito_db ... done
Creating pgadmin      ... done
--> Postgres ready.

Applying migrations...

psql: error: FATAL:  role "postgres" does not exist
psql: error: FATAL:  role "postgres" does not exist
psql: error: FATAL:  role "postgres" does not exist
psql: error: FATAL:  role "postgres" does not exist
psql: error: FATAL:  role "postgres" does not exist
Flyway is up to date
Flyway Teams Edition 8.0.3 by Redgate
Database: jdbc:postgresql://localhost:5432/vulcanito (PostgreSQL 13.3)
----------------------------------------
Flyway Teams features are enabled by default for the next 27 days. Learn more at https://rd.gt/3A4IWym
----------------------------------------
Successfully validated 49 migrations (execution time 00:00.137s)
Creating Schema History table "public"."flyway_schema_history" ...
Current version of schema "public": << Empty Schema >>
Migrating schema "public" to version "1.0 - initial schema"
Migrating schema "public" to version "1.2 - add scans fields"
Migrating schema "public" to version "1.3 - update asset types"
Migrating schema "public" to version "1.4 - add recipients table"
Migrating schema "public" to version "1.5 - permanent users"
Migrating schema "public" to version "1.6 - create table reports"
Migrating schema "public" to version "1.7 - add aws assettype"
Migrating schema "public" to version "1.9 - add partial unique constraint to recipients"
Migrating schema "public" to version "1.10 - add partial unique constraint to groups"
Migrating schema "public" to version "1.11 - add tag to teams"
Migrating schema "public" to version "1.12 - add partial unique constraint to assets"
Migrating schema "public" to version "1.13 - add ip range type"
Migrating schema "public" to version "1.14 - drop table scans"
Migrating schema "public" to version "1.15 - add autosend to programs"
Migrating schema "public" to version "1.16 - add program name to reports"
Migrating schema "public" to version "1.17 - add risk to reports"
Migrating schema "public" to version "1.18 - add docker image type"
Migrating schema "public" to version "1.19 - add web address type"
Migrating schema "public" to version "1.20 - users rename disabled to active"
Migrating schema "public" to version "1.21 - team members remove is default field"
Migrating schema "public" to version "1.22 - add user defaults and constraints"
Migrating schema "public" to version "1.23 - add asset scannable default and initialize"
Migrating schema "public" to version "1.24 - add description to groups"
Migrating schema "public" to version "1.25 - programs multpliple polices groups"
Migrating schema "public" to version "1.26 - add team id to programs"
Migrating schema "public" to version "1.27 - add grobal programs metadata"
Migrating schema "public" to version "1.28 - add description to policies"
Migrating schema "public" to version "1.29 - remove soft delete"
Migrating schema "public" to version "1.30 - add missing foreign keys and recreate current ones"
Migrating schema "public" to version "1.31 - add missing not null field constraints"
Migrating schema "public" to version "1.32 - audit table"
Migrating schema "public" to version "1.33 - security user"
Migrating schema "public" to version "1.34 - add git assettype"
Migrating schema "public" to version "1.35 - add user login"
Migrating schema "public" to version "1.36 - add observer to user"
Migrating schema "public" to version "1.37 - add rolfp to assets"
Migrating schema "public" to version "1.38 - add default rolfp value to assets"
Migrating schema "public" to version "1.39 - update default rolfp value for existing aws accounts"
Migrating schema "public" to version "1.40 - last classified at"
Migrating schema "public" to version "1.41 - add alias to assets"
Migrating schema "public" to version "1.42 - add enabled field to programs table"
Migrating schema "public" to version "1.43 - rename enabled to disabled in programs table"
Migrating schema "public" to version "1.44 - add custom schedule for global programs"
Migrating schema "public" to version "1.45 - drop reports"
Migrating schema "public" to version "1.46 - update teams unique tag"
Migrating schema "public" to version "1.47 - create outbox table"
Migrating schema "public" to version "1.48 - add disabled for global programs table"
Migrating schema "public" to version "1.49 - create findings overwrites"
Migrating schema "public" to version "1.50 - create asset annotations"
Successfully applied 49 migrations to schema "public", now at version v1.50 (execution time 00:02.072s)
Flyway is up to date
ERROR: Unable to obtain connection from database (jdbc:postgresql://localhost:5432/vulcanito_test) for user 'vulcanito_test': FATAL: password authentication failed for user "vulcanito_test"
--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
SQL State  : 28P01
Error Code : 0
Message    : FATAL: password authentication failed for user "vulcanito_test"

Caused by: org.postgresql.util.PSQLException: FATAL: password authentication failed for user "vulcanito_test"
--> Migrations applied successfully.

--> You can start the Vulcan API now.
```

### CODE REVIEW

File: contrib/keycloak/vulcan-realm/vulcan-realm.json [hunk 1]
```diff
@@ -0,0 +1,959 @@
...
+  },
+  "keycloakVersion" : "15.0.2",
+  "userManagedAccessAllowed" : false,
+  "clientProfiles" : {
+    "profiles" : [ ]
+  },
+  "clientPolicies" : {
+    "policies" : [ ]
+  }
+}
```
Comment (julianvilas) [inline contrib/keycloak/vulcan-realm/vulcan-realm.json:959]: Silly thing, but missing `\n` at the end of the file.

File: contrib/keycloak/vulcan-realm/vulcan-users-0.json [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+      "secretData" : "{\"value\":\"P/iMB7REi+lT+r9Cx5hoUGbgUY+npSAVeLuZ7yGuR9CxgBOkWhnHY5cTdZFs5n/0rLXqgFrcwexPAHPb0EZJNg==\",\"salt\":\"gD79MBxMsSHrY3FwiDPJdQ==\",\"additionalParameters\":{}}",
+      "credentialData" : "{\"hashIterations\":27500,\"algorithm\":\"pbkdf2-sha256\",\"additionalParameters\":{}}"
+    } ],
+    "disableableCredentialTypes" : [ ],
+    "requiredActions" : [ ],
+    "realmRoles" : [ "default-roles-vulcan" ],
+    "notBefore" : 0,
+    "groups" : [ ]
+  } ]
+}
```
Comment (julianvilas) [inline contrib/keycloak/vulcan-realm/vulcan-users-0.json:26]: Silly thing, but missing `\n` at the end of the file.

**Review by julianvilas — CHANGES_REQUESTED** [review]
LGTM, just a couple of silly comments to avoid files being concatenated when using shell commands.

Also I see that there are some conflicts to be resolved.

File: contrib/keycloak/vulcan-realm/vulcan-realm.json (hunk 1 above)
Comment (filewalkwithme) [reply contrib/keycloak/vulcan-realm/vulcan-realm.json]: Added a new line to the EOF!

File: contrib/keycloak/vulcan-realm/vulcan-users-0.json (hunk 2 above)
Comment (filewalkwithme) [reply contrib/keycloak/vulcan-realm/vulcan-users-0.json]: Added a new line to the EOF

**Comment by filewalkwithme** [comment]
> LGTM, just a couple of silly comments to avoid files being concatenated when using shell commands.
> 
> Also I see that there are some conflicts to be resolved.

Thanks! I have fixed the conflicts and also appended new lines at the end of the JSON files

**Review by julianvilas — APPROVED** [review]
LSGTM! Thanks a lot for the contribution @filewalkwithme :)

**Pull request merged**